Commented the decompressors (and some misc things)

GabrielRavier opened the pull request:
I think I've done them all. Kosinski, Enigma and Nemesis documentation all commented in their entirity (though I've partially copied Sega Retro on that one + the Sonic 1 disasm comments)

I've also commented 2-3 other things, the controller reading for exemple.

I'll add that if I made errors (or you think my comments blocks are too big), you can actually correct them as I've allowed edits from maintainers.

Clownacy commented on s2.asm:
```diff
@@ -1232,8 +1232,8 @@ Joypad_Read:
-	lea	(VDP_control_port).l,a0
-	lea	(VDP_data_port).l,a1
+	lea	(VDP_control_port).l,a0	; load VDP control port into a0
+	lea	(VDP_data_port).l,a1	; load VDP data port into a1
```
Seems pretty pointless.

Clownacy commented on s2.asm:
```diff
@@ -1767,54 +1787,69 @@ NemDec_WriteAndAdvance_XOR:
+; so the code needs to be bit-shifted appropriately over here before being used as a code table index
+; additionally, the code needs multiple entries in the table because no masking is done during compressed data processing
+; so if 11000 is a valid code then all indices of the form 11000XXX need to have the same entry
 loc_1606:
```
Legacy labels should be commented-out.

Clownacy commented on s2.asm:
```diff
@@ -2293,88 +2335,99 @@ EniDec_ChkGetNextByte:
+	lsr.w	#1,d5	; bit which is shifted out goes into C flag
 	move	sr,d6
-	dbf	d4,+
+	dbf		d4,Kos_Decomp_ChkBit
```
As s2.notes.txt says, the disassembly was set up with a tab size of 8 in mind, so this appears wrong.

GabrielRavier replied:
I though it looked fine but does that mean that ; =============== S U B R O U T I N E ======================================= appears wrong with 8 tabs ?

GabrielRavier replied:
Oh sorry I though window size not tabulation size sorry

LukeUsher commented on s2.asm:
```diff
@@ -176,7 +176,7 @@ ErrorTrap:
 EntryPoint:
 	tst.l	(HW_Port_1_Control-1).l	; test ports A and B control
-	bne.s	PortA_Ok	; If so, branch.
+	bne.s	PortA_Ok	; If so, magically branch.
```
Why magically? I believe this just checks if the IO ports have been initialised already, and skips the init process if so, nothing magic about it.

Clownacy commented on s2.asm:
```diff
@@ -2293,88 +2335,99 @@ EniDec_ChkGetNextByte:
+	lsr.w	#1,d5	; get next bit
 	move	sr,d6
-	dbf	d4,+
+	dbf		d4,Kos_Decomp_ChkBit2
```
Same here.

Clownacy commented on s2.asm:
```diff
@@ -2293,88 +2335,99 @@ EniDec_ChkGetNextByte:
+	move	d6,ccr	; was the bit set?
+	bcs.s	Kos_Decomp_FullMatch	; if it was, branch
+	lsr.w	#1,d5	; bit which is shifted out goes into X flag
+	dbf		d4,+
```
And here.

Clownacy commented on s2.asm:
```diff
@@ -2293,88 +2335,99 @@ EniDec_ChkGetNextByte:
+	roxl.w	#1,d3	; get high repeat count bit (shift X flag in)
 	lsr.w	#1,d5
-	dbf	d4,+
+	dbf		d4,+
```
This too.

Clownacy commented on s2.asm:
```diff
@@ -2293,88 +2335,99 @@ EniDec_ChkGetNextByte:
-	dbf	d3,-
-	bra.s	--
+	move.b	d0,(a1)+	; copy appropriate byte
+	dbf		d3,Kos_Decomp_MatchLoop	; and repeat the copying
```
This.

kirjavascript commented on s2.asm:
```diff
@@ -798,9 +798,9 @@ SS_PNTA_Transfer_Table:	offsetTable
-	subq.w	#1,(Demo_Time_left).w
+	tst.w	(Demo_Time_left).w	; is there time left on the demo?
+	beq.w	+	; if not, return
+	subq.w	#1,(Demo_Time_left).w	; subtract 1 from time left in demo
```
shouldn't comments that are basically;
`x = x + 1; // add 1 to x`
be omitted?

Clownacy replied:
If my memory's right, the other instances of this code already had comments like these, so it's more of a consistency thing.

Granted, I would prefer this disassembly have little of its fat trimmed, but that is a lot of code to sift through. You also have the fact that coding novices read through this.

GabrielRavier commented:
I corrected all my errors you pointed out

flamewing commented:
And again with a generic commit message that does not have any relation to what the commit does.

GabrielRavier commented:
I didn't remember exactly what were the contents of the changes. Adding actual commit message...

GabrielRavier commented:
Also here is the complete changelog : 
1. Made a few changes to the header comments
2. "4 bytes" to "a long"
3. Removed things such as "subtract 1 from time left in demo"
4. Reformulated a few comments
5. Commented the controller-reading
6. Removed a tab in an empty space
7. Commented the Nemesis decompression routine
8. Commented the Enigma decompression routine
9. Commented the Kosinski decompression routine
10. Changed strange commenting in Super Sonic PAL cycle

GabrielRavier commented:
Fixed that commit Clownacy

GabrielRavier commented:
Closing this because I have no intent to correct this ever and I also am preparing another pull request so...
